client/src/main/java/com/netflix/conductor/client/http/TaskClient.java

```diff
@@ -210,25 +243,37 @@ public Task getPendingTaskForWorkflow(String workflowId, String taskReferenceNam
     /**
      * Updates the result of a task execution.
      *
-     * @param taskResult TaskResults to be updated.
-     * @param taskType
+     * @param taskResult the {@link TaskResult} of the executed task to be updated.
+     * @param taskType   the type of the task
      */
     public void updateTask(TaskResult taskResult, String taskType) {
         Preconditions.checkNotNull(taskResult, "Task result cannot be null");
+        Preconditions.checkArgument(StringUtils.isBlank(taskResult.getExternalOutputPayloadStoragePath()), "External Storage Path must not be set");
 
-        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-             CountingOutputStream countingOutputStream = new CountingOutputStream(byteArrayOutputStream)) {
-            objectMapper.writeValue(countingOutputStream, taskResult);
-            long taskResultSize = countingOutputStream.getCount();
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+            objectMapper.writeValue(byteArrayOutputStream, taskResult.getOutputData());
+            byte[] taskOutputBytes = byteArrayOutputStream.toByteArray();
+            long taskResultSize = taskOutputBytes.length;
             WorkflowTaskMetrics.recordTaskResultPayloadSize(taskType, taskResultSize);
-            if (taskResultSize > (3 * 1024 * 1024)) { //There is hard coded since there is no easy way to pass a config in here
-                taskResult.setReasonForIncompletion(String.format("The TaskResult payload: %d is greater than the permissible 3MB", taskResultSize));
-                taskResult.setStatus(TaskResult.Status.FAILED_WITH_TERMINAL_ERROR);
-                taskResult.setOutputData(null);
+
+            long payloadSizeThreshold = conductorClientConfiguration.getTaskOutputPayloadThresholdKB() * 1024;
+            if (taskResultSize > payloadSizeThreshold) {
+                if (!conductorClientConfiguration.isExternalPayloadStorageEnabled()
+                        || taskResultSize > (conductorClientConfiguration.getTaskOutputMaxPayloadThresholdKB() * 1024)) {
```

Comment:
Do you need extra brackets?  Is it easier to deal with kb everywhere?

Reply:
Added javadoc and removed brackets..